tutorials/basic-topics/sorting/shell-sort.md

```diff
@@ -11,4 +10,463 @@ keywords:
+
+                # case 2: both pointers in nums2[]
+                elif left >= m:
+                    self.swapIfGreater(nums2, nums2, left - m, right - m)
+
+                # case 3: both pointers in nums1[]
+                else:
+                    self.swapIfGreater(nums1, nums1, left, right)
+
+                left += 1
+                right += 1
+
+            # Break if iteration with gap=1 is completed
+            if gap == 1:
+                break
+
+            # Calculate the new gap
+            gap = (gap / 2) + (gap % 2)
+
+        j = 0
+        for i in range(m, len):
+            nums1[i] = nums2[j]
+            j += 1
+```
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+class Solution {
+    public void swapIfGreater(int[] arr1, int[] arr2, int ind1, int ind2) {
+        if (arr1[ind1] > arr2[ind2]) {
+            int temp = arr1[ind1];
+            arr1[ind1] = arr2[ind2];
+            arr2[ind2] = temp;
+        }
+    }
+
+    public void merge(int[] nums1, int m, int[] nums2, int n) {
+        int len = m + n;
+
+        // Initial gap
+        int gap = (len / 2) + (len % 2);
+
+        while (gap > 0) {
+            // Place two pointers
+            int left = 0;
+            int right = left + gap;
+
+            while (right < len) {
+                // case 1: left in nums1[] and right in nums2[]
+                if (left < m && right >= m) {
+                    swapIfGreater(nums1, nums2, left, right - m);
+                }
+                // case 2: both pointers in nums2[]
+                else if (left >= m) {
+                    swapIfGreater(nums2, nums2, left - m, right - m);
+                }
+                // case 3: both pointers in nums1[]
+                else {
+                    swapIfGreater(nums1, nums1, left, right);
+                }
+                left++;
+                right++;
+            }
+            // Break if iteration with gap=1 is completed
+            if (gap == 1) {
+                break;
+            }
+            // Calculate the new gap
+            gap = (gap / 2) + (gap % 2);
+        }
+
+        int j = 0;
+        for (int i = m; i < len; i++) {
+            nums1[i] = nums2[j];
+            j++;
+        }
+    }
+}
+```
+</TabItem>
+</Tabs>
+
+**Time Complexity**: $O((n+m) \cdot \log(n+m))$
+
+The time complexity of the algorithm for merging two sorted arrays of sizes $n$ and $m$ is determined as follows:
+1.
+- The gap ranges from $n+m$ to 1, and at each step, the gap is divided by 2.
+- Therefore, the outer loop runs for $O(log(n+m))$ iterations.
+
+For each value of the gap, the inner loop can run for a maximum of $n+m$ times.
+
+**Space Complexity** : $O(1)$
+
+**Efficiency Consideration**
+
+In this question we can use brute approach of merging 2 sorted arrays using extra space but for optimal solution where we do not consider any extra space Shell Sort gives us an efficient solution.
+
+export const suggestedProblems = [
```

Comment:
for each `problemName`, please add a space between `-`.  Also pad 0, e.g. `912` -> `0912`, `75` -> `0075`, etc.

Reply:
space one not resolved. e.g. `0912-Sort an Array` -> `0912 - Sort an Array`

Reply:
What is the reason to delete 0912?

Reply:
As it is already given in the examples